src/helper/service.cpp

```diff
@@ -72,6 +77,36 @@ service::ptr service::from_settings(const service_identifier &id,
         std::chrono::milliseconds{rc_settings.poll_interval});
 }
 
+void service::handle_error()
+{
+    rc_action_ = [this] { discover(); };
+    errors_++;
```

Comment:
This will overflow and go negative, what happens then? Perhaps you want to change this to a `uintXX_t` and also limit the maximum number of errors.

Reply:
Ok